src/borg/repository.py

```diff
@@ -396,7 +396,8 @@ def open(self, path, exclusive, lock_wait=None, lock=True):
         try:
             st = os.stat(path)
         except FileNotFoundError:
-            raise self.DoesNotExist(path)
+            logger.warning("%s is not a valid borgbackup directory." % (path))
+            os.exit(2)
```

Comment:
this does not seem to fix the issue you referred to, but it changes the behaviour if there is no repository directory at all.

not sure the behaviour change is the right way either, see the other places below where it handles misc. error conditions also by raising a specific exception, so why should it be different here?

in general, you need to reproduce the issue without any change, then fix it and try to reproduce it again (or even better: add a test, so we can automatically test that).

for repository code changes, you also need to test with a local repo and a ssh: repo (borg in client/server mode) and make sure both scenarious work ok.